.buildkite/scripts/sbt_dependency_tree.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+# -*- encoding: utf-8
+
+import json
+import os
+
+
+class Project:
+    def __init__(self, name, folder, dependencies):
+        self.name = name
+        self.folder = folder
+        self.dependencies = dependencies
+
+    def __repr__(self):
+        return "<%s name=%r folder=%r dependencies=%r>" % (
+            type(self).__name__,
+            self.name,
+            self.folder,
+            ",".join(sorted(d.name for d in self.dependencies)),
+        )
+
+    def all_folders(self):
+        folders = set([self.folder])
+        for d in self.dependencies:
+            folders |= d.all_folders()
+        return folders
+
+    def depends_on(self, other):
+        if self == other:
+            return True
+        if other in self.dependencies:
+            return True
+        for d in self.dependencies:
+            return d.depends_on(other)
```

Comment:
Hmm this will return on the first iteration of the loop, is this what you actually want?

Reply:
Maybe it should be more like this?

```python
return (
    self == other or
    other in self.dependencies or
    any(d.depends_on(other) for d in self.dependencies))
)
```

Reply:
This is copied without modification from the Travis build, so i'm not certain (other than that this will reproduce existing behaviour). Perhaps @alexwlchan can shed some light.

Reply:
Yeah, this is probably a bug.